utils/test_utils.py

```diff
@@ -69,7 +69,8 @@ def try_lossless_compression(
     output_bits_block = encoder.encode_block(data_block)
 
     # test decode
-    decoded_block, _ = decoder.decode_block(output_bits_block)
+    decoded_block, num_bits_consumed = decoder.decode_block(output_bits_block)
+    assert num_bits_consumed == len(output_bits_block)
```

Comment:
Nice. Wonder how we missed this before

Reply:
actually want to improve this test..  by optionally adding some extra bits to encoder output. This is to simulate the scenario where there are additional bits beyond the bitstream encoded. Some compressors (like range coder, AEC) support this .

Reply:
@shubhamchandak94: take a re-look